components/selectors/context.rs

```diff
@@ -181,6 +181,12 @@ where
         }
     }
 
+    /// How many times deep are we in a selector.
+    #[inline]
+    pub fn nesting_level(&self) -> usize {
```

Comment:
Maybe consider introducing a `is_nested` method instead, given we only call this to compare it to 0.